src/main/java/org/spongepowered/common/data/processor/value/entity/IsFlyingValueProcessor.java

```diff
@@ -0,0 +1,98 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.value.entity;
+
+import com.google.common.base.Optional;
+import net.minecraft.entity.player.EntityPlayer;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.ImmutableDataCachingUtil;
+import org.spongepowered.common.data.processor.common.AbstractSpongeValueProcessor;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+import org.spongepowered.common.data.value.mutable.SpongeValue;
+
+public class IsFlyingValueProcessor extends AbstractSpongeValueProcessor<Boolean, Value<Boolean>> {
+
+    protected IsFlyingValueProcessor() {
+        super(Keys.IS_FLYING);
+    }
+
+    @Override
+    public Optional<Boolean> getValueFromContainer(ValueContainer<?> container) {
+        if (supports(container)) {
+            net.minecraft.entity.Entity entity = ((net.minecraft.entity.Entity) container);
+            boolean isFlying = entity.isAirBorne;
+            return Optional.of(isFlying);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public boolean supports(ValueContainer<?> container) {
+        return container instanceof net.minecraft.entity.Entity;
+    }
+
+    @Override
+    public DataTransactionResult offerToStore(ValueContainer<?> container, Boolean value) {
+        final ImmutableValue<Boolean> proposedValue = ImmutableDataCachingUtil.getValue(ImmutableSpongeValue.class, Keys.IS_FLYING, value, false);
+        if (supports(container)) {
+            final DataTransactionBuilder builder = DataTransactionBuilder.builder();
+            final ImmutableValue<Boolean> newFlyingValue =
+                    ImmutableDataCachingUtil.getValue(ImmutableSpongeValue.class, Keys.IS_FLYING, value, false);
+            final ImmutableValue<Boolean> oldFlyingData = getApiValueFromContainer(container).get().asImmutable();
+            if (value == null) {
+                return DataTransactionBuilder.errorResult(newFlyingValue);
+            }
+            try {
+                net.minecraft.entity.Entity entity = ((net.minecraft.entity.Entity) container);
+                if (entity instanceof EntityPlayer) {
+                    EntityPlayer player = (EntityPlayer) entity;
+                    player.capabilities.allowFlying = newFlyingValue.get();
+                } else {
+                    entity.isAirBorne = newFlyingValue.get();
+                }
+                return builder.success(newFlyingValue).replace(oldFlyingData).result(DataTransactionResult.Type.SUCCESS).build();
```

Comment:
You could just do `return DataTransactionBuilder.successReplace(oldFlyingData, newFlyingValue);`